foodtruck/foodtruck.go

```diff
@@ -83,36 +107,39 @@ func Listen() {
 func receive(o []byte) {
 	order := Order{}
 	json.Unmarshal(o, &order)
-	fmt.Printf("Order %v Received! ", order.ID)
-}
-
-func Send() {
-	order := Order{
-		ID:       "1",
-		Policies: []Policy{},
-		Change: Change{
-			Ticket:      "abc123",
-			WindowStart: time.Now(),
-			WindowStop:  time.Date(2020, 12, 31, 0, 0, 0, 0, time.Local),
-		},
-	}
-	jsonOrder, err := json.Marshal(order)
+	fmt.Printf("Order %v Received!\n", order.ID)
+	err := ioutil.WriteFile(".foodtruck/orders/"+order.ID+".json", o, 0700)
 	if err != nil {
 		panic(err)
 	}
-	err = c.SendOrder(jsonOrder)
+	processOrder(order)
+}
+
+func Send() {
+	order := `{"id":"` + uuid.New().String() + `","policies":[{"provider":"mock","definition":{"attrib1":"abc","attrib2":"123","nested":{"attrib3":"a1"}}}],"change":{"ticket":"abc123","start":"2020-01-01 00:00:00", "end":"2021-01-01 00:00:00"}}`
+	err := c.SendOrder([]byte(order))
 	if err != nil {
 		panic(err)
 	}
 }
 
-func ProcessOrder(o Order) {
-	for _, p := range o.Policies {
-		switch p.Provider {
+func processOrder(o Order) {
+	var p Provider
+
+	for _, policy := range o.Policies {
+		switch policy.Provider {
 		case "chefinfra":
-			chefinfra.Execute(p.Definition)
+			ensureDir(".foodtruck/chefinfra")
+			p = chefinfra.New(policy.Definition)
 		case "chefinspec":
-			chefinspec.Execute(p.Definition)
+			ensureDir(".foodtruck/chefinspec")
+			p = chefinspec.New(policy.Definition)
+		case "mock":
+			p = mock.New()
```

Comment:
Should we have something like: 

```
else
    fmt.printf("The provider specified is not supported")
    panic(err)
```

Reply:
I think that would be a really good idea.  Do you want to add that in?